### PR TITLE
Hyperchunk2d

### PR DESCRIPTION
**Pull request by jreadey**
This PR adds hyperchunking support for multi-dimensional datasets.

### CODE REVIEW

File: hsds/dset_lib.py [hunk 1]
```diff
@@ -66,6 +67,41 @@ def getFillValue(dset_json):
...
+        pt[dim] = (index * factor + i)
+        if dim + 1 == rank:
+            idx = int(arr_index) + i
+            if rank == 1:
+                arr_points[idx] = pt[0]  # need to set 1d arrays with an int index
+            else:
+                arr_points[idx] = pt
+        else:
+            kwargs = {"dim": dim + 1, "chunk_index": chunk_index, "factors": factors}
+            next_index = arr_index + (i * np.prod(factors[1:]))
```
Comment (mattjala) [inline hsds/dset_lib.py:101]: Should this use `np.prod(factors[i:]` to account for all remaining dimensions? This might not matter right now since hyperchunking is limited to two dimensions at most, but it would be easier to understand and future-proof

File: hsds/util/rangegetUtil.py [hunk 2]
```diff
@@ -20,6 +21,37 @@ def _chunk_start(c):
     return start
 
 
+def getHyperChunkFactors(chunk_dims, hyper_dims):
+    """ return list of rations betwen chunk and hyperchunkdims """
```
Comment (mattjala) [inline hsds/util/rangegetUtil.py:25]: `rations` -> `ratios`

File: hsds/dset_lib.py [hunk 3]
```diff
@@ -66,6 +67,41 @@ def getFillValue(dset_json):
     return arr
 
 
+def _get_arr_pts(arr_points, arr_index, pt, dim=0, chunk_index=None, factors=None):
+    """ recursive function to fill in chunk locations for hyperchunk selection.
+        arr_points: numpy array of shape (num_chunks, rank)
```
Comment (mattjala) [inline hsds/dset_lib.py:72]: Might be more clear to reword this to something like `Recursive function which fills arr_points with the coordinates to locate a hyperchunk selection's corresponding chunks in a chunk table`.

File: hsds/chunk_dn.py [hunk 4]
```diff
@@ -384,6 +382,16 @@ async def GET_Chunk(request):
...
+    """
+    if "hyper_index" in params:
+        param_hyper_index = params["hyper_index"]
+        log.debug(f"tbd param_hyper_index: {param_hyper_index}")
+        if param_hyper_index.find(":") > 0:
+            hyper_index = param_hyper_index.split(":")
+        else:
+            hyper_index = param_hyper_index
+        log.debug(f"hyper_index: {hyper_index}")
+    """
```
Comment (mattjala) [inline hsds/chunk_dn.py:394]: Unused

**Review by mattjala — DISMISSED** [review]
LGTM. The integration test failures are due to #311

File: hsds/dset_lib.py (hunk 1 above)
Comment (jreadey) [reply hsds/dset_lib.py:101]: Yes this is a bug that shows up with more than two dimensions.  Fixed this and add a 3D test.  Should work for any number of dimensions now

File: hsds/util/rangegetUtil.py (hunk 2 above)
Comment (jreadey) [reply hsds/util/rangegetUtil.py:25]: Fixed

File: hsds/dset_lib.py (hunk 3 above)
Comment (jreadey) [reply hsds/dset_lib.py:72]: Sounds better. Updated

File: hsds/chunk_dn.py (hunk 4 above)
Comment (jreadey) [reply hsds/chunk_dn.py:394]: Removed
